### PR TITLE
Add error line cache and counter to bug suggestions.

### PR DESCRIPTION
**Pull request by jmaher**

### CODE REVIEW

**Comment by jmaher** [comment]
as a note, this will just do the backend, to keep the front end code simple, I will add front end changes in early october to take advantage of this.

File: treeherder/model/error_summary.py [hunk 1]
```diff
@@ -33,6 +35,19 @@ def get_error_summary(job, queryset=None):
...
+    line_cache_key = 'error_lines'
+    line_cache = cache.get(line_cache_key)
+    if line_cache is None:
+        line_cache = {str(job.submit_time.date()): {}}
+    else:
+        dates = list(line_cache.keys())
+        dates.sort()
+        for d in dates:
+            dTime = datetime.datetime.strptime(d, '%Y-%m-%d')
+            if dTime <= (job.submit_time - datetime.timedelta(days=LINE_CACHE_TIMEOUT_DAYS)):
```
Comment (Archaeopteryx) [inline treeherder/model/error_summary.py:48]: This applies only once per day. An `else` condition which adds a `break` statement would be beneficial.

File: treeherder/model/error_summary.py [hunk 2]
```diff
@@ -42,18 +57,32 @@ def get_error_summary(job, queryset=None):
...
 
     try:
         cache.set(cache_key, error_summary, BUG_SUGGESTION_CACHE_TIMEOUT)
     except Exception as e:
         newrelic.agent.record_custom_event('error caching error_summary for job', job.id)
         logger.error('error caching error_summary for job %s: %s', job.id, e, exc_info=True)
 
+    try:
+        cache.set(
+            line_cache_key, line_cache, BUG_SUGGESTION_CACHE_TIMEOUT * LINE_CACHE_TIMEOUT_DAYS
```
Comment (Archaeopteryx) [inline treeherder/model/error_summary.py:76]: This should not rely on `BUG_SUGGESTION_CACHE_TIMEOUT` being a day but use a separate constant for this.

File: treeherder/model/error_summary.py [hunk 3]
```diff
@@ -65,9 +94,31 @@ def bug_suggestions_line(err, term_cache=None):
...
+        today = str(logdate.date())
+    if today not in line_cache.keys():
+        line_cache[today] = {}
+
     # remove the mozharness prefix
     clean_line = get_cleaned_line(err.line)
 
+    # remove floating point numbers from the summary as they are often variable
+    cache_clean_line = re.sub(r' [0-9]+\.[0-9]+ ', ' X ', clean_line)
+    cache_clean_line = re.sub(r' leaked [0-9]+ window(s)', ' leaked X window(s)', cache_clean_line)
```
Comment (Archaeopteryx) [inline treeherder/model/error_summary.py:111]: Is the number of leaked windows random?

**Review by Archaeopteryx — APPROVED** [review]

File: treeherder/model/error_summary.py (hunk 3 above)
Comment (jmaher) [reply treeherder/model/error_summary.py:111]: I see 1,2,3,4 in my 5 weeks of downloaded failure data.

**Pull request merged**